src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts

```diff
@@ -0,0 +1,269 @@
+import { BranchPathAuthoringComponent } from '../branch-path-authoring/branch-path-authoring.component';
+import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
+import { TeacherProjectService } from '../../services/teacherProjectService';
+
+@Component({
+  imports: [
+    BranchPathAuthoringComponent,
+    FlexLayoutModule,
+    MatButtonModule,
+    ReactiveFormsModule,
+    RouterModule,
+    SelectBranchCriteriaComponent,
+    SelectComponentComponent,
+    SelectStepComponent
+  ],
+  selector: 'edit-branch',
+  standalone: true,
+  styleUrl: './edit-branch.component.scss',
+  templateUrl: './edit-branch.component.html'
+})
+export class EditBranchComponent extends AbstractBranchAuthoringComponent {
+  private branchPaths: any[] = [];
+  protected mergeStepTitle: string;
+  private node: any;
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    protected fb: FormBuilder,
+    protected projectService: TeacherProjectService,
+    protected route: ActivatedRoute,
+    protected router: Router
+  ) {
+    super(fb, projectService, route, router);
+  }
+
+  ngOnInit(): void {
+    super.ngOnInit();
+    const branch = this.projectService.getBranchesByBranchStartPointNodeId(this.targetId)[0];
+    this.setPathCount(branch.paths.length);
+    this.setMergeStep(branch.endPoint);
+    this.mergeStepTitle = this.projectService.getNodePositionAndTitle(branch.endPoint);
+    this.node = this.projectService.getNodeById(this.targetId);
+  }
+
+  ngAfterViewInit(): void {
+    this.populateBranchAuthoring();
+  }
+
+  private populateBranchAuthoring(): void {
+    this.branchPaths = this.getBranchPaths();
+    const criteria = this.getBranchCriteria();
+    this.setCriteria(criteria);
+    const branchParamNodeId = this.getBranchParamByField('nodeId');
+    if (branchParamNodeId != null) {
+      this.setNodeId(branchParamNodeId);
+    }
+    const branchParamComponentId = this.getBranchParamByField('componentId');
+    if (branchParamComponentId != null) {
+      this.setComponentId(branchParamComponentId);
+    }
+    this.updateParamFormControls(criteria);
+    this.changeDetector.detectChanges();
+  }
+
+  private getBranchPaths(): any[] {
+    const branchPaths = [];
+    for (let t = 0; t < this.node.transitionLogic.transitions.length; t++) {
+      const transition = this.node.transitionLogic.transitions[t];
+      const branch: any = {
+        number: t + 1,
+        items: this.getItems(),
+        nodesInBranchPath: [],
+        transition: transition
+      };
+      branchPaths.push(branch);
+      this.populateBranchParams(branch, transition);
+    }
+    return branchPaths;
+  }
+
+  private populateBranchParams(branch: any, transition: any): void {
+    if (transition.criteria != null) {
+      for (const criterion of transition.criteria) {
+        if (criterion.name === this.SCORE_VALUE) {
+          this.populateScoreBranchParam(branch, criterion);
```

Comment:
Do we need to call a function here?
```suggestion
          branch.scores = criterion.params.scores;
```